### PR TITLE
Parsed model

### PR DESCRIPTION
**Pull request by eneufeld**
Changed code to use a "Model" instead of the JsonSchema directly

### CODE REVIEW

File: src/path.util.ts [hunk 1]
```diff
@@ -47,45 +47,19 @@ export const getValuePropertyPair = (instance: any, path: string):
...
-  }
+    (pathSegment) => pathSegment === '#' || pathSegment === undefined || pathSegment === '' ||
+    pathSegment === 'properties';
+  const resultSchema = validPathSegments.reduce((curModel, pathSegment) =>
+    invalidSegment(pathSegment) ? curModel :
+      isItemModel(curModel) ?
+        curModel.dropPoints.hasOwnProperty(pathSegment) ? curModel.dropPoints[pathSegment]
+          : curModel.attributes[pathSegment]
+        : curModel, model);
+  // if (resultSchema !== undefined && resultSchema.$ref !== undefined) {
```
Comment (edgarmueller) [inline src/path.util.ts]: Remove commented code

File: src/renderers/additional/table-array.control.ts [hunk 2]
```diff
@@ -72,11 +73,11 @@ export class TableArrayControlRenderer extends Renderer implements DataChangeLis
...
     const head = document.createElement('thead');
     const headRow = document.createElement('tr');
-    const resolvedSchema = resolveSchema(this.dataSchema, controlElement.scope.$ref + '/items');
-    Object.keys(resolvedSchema.properties).forEach(key => {
-      if (resolvedSchema.properties[key].type === 'array') {
-        return;
-      }
+    const resolvedSchema = <ItemModel>resolveSchema(this.dataModel, controlElement.scope.$ref);
+    Object.keys(resolvedSchema.attributes).forEach(key => {
+      // if (resolvedSchema.properties[key].type === 'array') {
```
Comment (edgarmueller) [inline src/renderers/additional/table-array.control.ts]: Remove commented code

File: src/renderers/additional/table-array.control.ts [hunk 3]
```diff
@@ -87,10 +88,10 @@ export class TableArrayControlRenderer extends Renderer implements DataChangeLis
     let arrayData = this.dataService.getValue(controlElement);
     const renderChild = (element: Object): void => {
       const row = document.createElement('tr');
-      Object.keys(resolvedSchema.properties).forEach(key => {
-        if (resolvedSchema.properties[key].type === 'array') {
-          return;
-        }
+      Object.keys(resolvedSchema.attributes).forEach(key => {
+        // if (resolvedSchema.properties[key].type === 'array') {
```
Comment (edgarmueller) [inline src/renderers/additional/table-array.control.ts]: Remove commented code

File: test/label.util.test.ts [hunk 4]
```diff
@@ -226,7 +240,8 @@ test('control with label object, text-only and required', t => {
     } as JsonSchema;
     const controlElement = {type: 'ControlElement', scope: {$ref: '#/properties/foo'},
       label: {text: 'mega bar'}} as ControlElement;
-    const labelObject = getElementLabelObject(schema, controlElement);
+    const labelObject = getElementLabelObject({schema: schema, dropPoints: {}} as ItemModel,
```
Comment (edgarmueller) [inline test/label.util.test.ts]: Extract creation of `ItemModel` into helper factory function, which makes it easy to update the initialization of the item. Also removes the necessity to cast.

File: test/renderers/array-renderer.test.ts [hunk 5]
```diff
@@ -93,7 +124,38 @@ test('generate array child control w/o data', t => {
     };
     const data = {};
     renderer.setDataService(new DataService(data));
-    renderer.setDataSchema(schema);
+    renderer.setDataModel({
+  schema: {type: 'object', properties: {test: {type: 'array', items: {
```
Comment (edgarmueller) [inline test/renderers/array-renderer.test.ts]: Why isn't the schema used we already have in scope? Applies to other tests as well, check for unused local vars.

**Comment by eneufeld** [comment]
I keep this open for the moment until we moved all the features to the schema service

**Comment by eneufeld** [comment]
This is outdated

**Pull request closed without merging**